src/persistence/history.cpp

```diff
@@ -26,10 +26,11 @@
 #include "db/rawdatabase.h"
 
 namespace {
-static constexpr int SCHEMA_VERSION = 1;
+static constexpr int SCHEMA_VERSION = 3;
 
-void generateCurrentSchema(QVector<RawDatabase::Query>& queries)
+bool createCurrentSchema(std::shared_ptr<RawDatabase>& db)
```

Comment:
Why do you pass the pointer by reference? It doesn't look like you change pointer itself. 
Since these functions only uses a pointer, it's should be a raw pointer. See [CppCoreGudlines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#r3-a-raw-pointer-a-t-is-non-owning)

Reply:
Changed to a raw pointer, rather than a raw reference due to the arguments made here https://google.github.io/styleguide/cppguide.html#Output_Parameters.

Reply:
1. Sorry, I meant raw reference.
2. Why `db` is an output parameter?
3. If you use the pointer, but not reference, it will be good to check, that `db` is not `nullptr`. Fiil free to ignore is point, because out of the scope.

Reply:
re 2., oops I linked to the wrong section, I meant to link to https://google.github.io/styleguide/cppguide.html#Reference_Arguments.
re 3:, I feel like both the copy option, reference option, and pointer options are bad =/
3.1 copy forces an atomic increment and copy of the smart pointer object
3.2 reference has the pointer implementation with force non-null that I want, but as mentioned in the corrected link "References can be confusing, as they have value syntax but pointer semantics."
3.3 pointer can be null, requiring either null checks on every small function, or weak function documents like `// @param db, pointer to database, can't be null` with no actual language support

stupid C++ =/

of these I prefer the pointer option, but dislike the boilerplate code required on every function to check pointers are non-null when they're basically already known to be non-null... I guess checking pre-conditions is still a good idea. I've added an assert non null in each function.

Reply:
Stupid not C++ (in this case, at least ;) ), but Google Cpp Guide. "References can be confusing, as they have value syntax but pointer semantics.". If I want to iterate over a container and change it's value I just use `for (auto& x : vector) { x.change(); } `. Nothing more, nothing less. Why I can't do the same with functions. 

I suggest using point from [CppCoreGudlines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f17-for-in-out-parameters-pass-by-reference-to-non-const). At a minimum, this is a more generally accepted guide, than GoogleCppGuid.

Reply:
I have no problem with it in a case like `for (auto& x : vector) { x.change(); }`, but for functions, it looks to the caller like they're passing in a copy of the argument, when they're really passing in a modifiable reference to it. Also when the reference is stored in a class long term, it looks again like you hold the object, when you might just have a non-owning reference to someone else's object. Pointers having their own syntax makes it clear every time you use `->` that you're not holding the object directly.

I think having a complete set of coding standards is more valuable than arguing each point, so I'm happy to yield to iso CPP core guidelines. I'll update these to references. I also chatted with @sudden6 and one of us will add a line to our coding standards to refer to cpp core guidelines in that case where something isn't covered by our coding standard (like this).